src/kagglehub/clients.py

```diff
@@ -44,6 +44,44 @@
 """
 
 KAGGLEHUB_USER_AGENT = {"User-Agent": f"kagglehub/{kagglehub.__version__}"}
+_cached_user_agent = None
+
+
+class KaggleHubUserAgent:
+    def __init__(self) -> None:
+        self._cached_user_agent: Optional[str] = None
+
+    def _is_in_kaggle_notebook(self) -> bool:
+        return os.getenv("KAGGLE_NOTEBOOK_ENV_VAR_NAME") is not None
+
+    def _is_in_colab_notebook(self) -> bool:
+        return os.getenv("TBE_RUNTIME_ADDR") is not None and os.getenv("COLAB_RELEASE_TAG") is not None
```

Comment:
For Colab, see the latest comment from Seb: https://b.corp.google.com/issues/322393193#comment5

Only check for `COLAB_RELEASE_TAG` and add `-managed/unmanaged` suffix based on the `TBE_RUNTIME_ADDR` runtime presence.